### PR TITLE
Implement new notification badge component

### LINKED ISSUE #644: Implement new notification badge component
### Problem

@sandrina-p noticed that the new UI did not have a notification badge specified yet.

### Solution

I have redesigned this component, with 2 variants, and added it to our [Figma design system](https://www.figma.com/file/mxGadAHfkWH6qApebQvcdN/Group-Income-2.0?node-id=1%3A7):
- Default size;
- Compact;

**Screenshots:**

Captura de ecrã 2019-08-29, às 16 48 38

Captura de ecrã 2019-08-29, às 16 45 18

Captura de ecrã 2019-08-29, às 16 31 49

Related to #636

### PR DESCRIPTION
**Pull request by SebinSong**
closes #644 
replaces the old badge with the new one.
the new 'Badge.vue' component expects the 'type' prop to be specified as 'compact' if the compact style is wanted.

### CODE REVIEW

**Review by mmbotelho — CHANGES_REQUESTED** [review]
Captura de ecrã 2019-09-07, às 15 13 21

Just a tiny detail - the number is slightly off-center vertically! ☝️ 

Otherwise, great job!

File: frontend/views/components/Badge.vue
Comment (sandrina-p) [inline frontend/views/components/Badge.vue]: Use `rem` unities here too. You can use `$spacer-xs`

File: frontend/views/components/ListItem.vue
Comment (sandrina-p) [inline frontend/views/components/ListItem.vue]: It's missing the `badgeCount` could here as children

Comment (sandrina-p) [inline frontend/views/components/ListItem.vue]: It's missing the badgeCount here as children

File: frontend/views/components/Badge.vue
Comment (sandrina-p) [inline frontend/views/components/Badge.vue]: Use rem unities here. You can use $spacer-xs

**Review by sandrina-p — CHANGES_REQUESTED** [review]

File: frontend/views/components/Badge.vue
Comment (sandrina-p) [inline frontend/views/components/Badge.vue:12]: It would be nice if you added this component to our Design System page as well! [http://localhost:8000/app/design-system](http://localhost:8000/app/design-system)

Comment (SebinSong) [reply frontend/views/components/Badge.vue:12]: I will definitely work on it. :) Thanks.

File: frontend/views/components/ListItem.vue
Comment (SebinSong) [reply frontend/views/components/ListItem.vue]: The 'compact' type doesn't show the count of notifications so I didn't pass it down here. :)
I was wondering though if passing down the count as a prop like in the old component is better than using a slot. What would you think? Thanks! :)

Comment (sandrina-p) [reply frontend/views/components/ListItem.vue]: Oh right, it makes sense visually. We must not forget to make it more accessible by adding to the span `aria-label="New {slot} notifications"`, so a user using screen reader knows about it.

Here I think we can not put a slot inside a string as we do with props, but I'm not sure. If we can, cool, otherwise we need to rollback it to a prop. Do this for both types of badge.
Last but not least give it a `role="alert"`. This will turn the element "alive" for screen readers. When the content changes, it lets the user know about it.

File: frontend/views/components/Badge.vue
Comment (sandrina-p) [reply frontend/views/components/Badge.vue:12]: Looks cool, well done!

File: frontend/views/components/ListItem.vue
Comment (SebinSong) [reply frontend/views/components/ListItem.vue]: It's a very informative comment..! :)
let me try using '$slots.default' for the aria-label and rolling back to a prop if it doesn't work.

**Review by mmbotelho — DISMISSED** [review]
Looking great! Thank you @SebinSong 🎉

File: frontend/views/components/Badge.vue
Comment (taoeffect) [inline frontend/views/components/Badge.vue]: Since this is a user-facing string, it too needs to pass through the `L` function. So I would recommend something like (typing this up from memory):

```
:aria-label='L("{num} notifications", { num: $slots.default ? $slots.default[0].text : 0 })'
```

Please double-check also via the browser inspector tool that this text is in fact being set correctly on this attribute.

**Review by taoeffect — CHANGES_REQUESTED** [review]
It looks great to me @SebinSong! Just one minor issue found related to localization for the `aria-label` text and once that's fixed has my approval.

File: frontend/views/components/Badge.vue
Comment (SebinSong) [reply frontend/views/components/Badge.vue]: Thanks greg. I will work on it soon. :)

**Review by sandrina-p — DISMISSED** [review]

**Comment by taoeffect** [comment]
@SebinSong just a heads up, with a recent merge there are now merge conflicts

File: frontend/views/components/Badge.vue
Comment (SebinSong) [reply frontend/views/components/Badge.vue]: Thanks greg! I will make sure to merge it before I work on your change request, maybe tonight!

File: frontend/assets/locales/en/translation.json
Comment (taoeffect) [inline frontend/assets/locales/en/translation.json:1]: Heh, you didn't need to update this file but this is OK. In the future don't update this file manually (it's something that will be auto-generated).

File: frontend/views/components/Badge.vue
Comment (taoeffect) [inline frontend/views/components/Badge.vue]: This needs to be:

```js
:aria-label='L("{num} new notifications", { num: $slots.default ? $slots.default[0].text : "" })'
```

**Review by taoeffect — CHANGES_REQUESTED** [review]
See comments

**Review by taoeffect — APPROVED** [review]
Awesome!

**Pull request merged**
